snowflake/connect.go

```diff
@@ -0,0 +1,70 @@
+package snowflake
+
+// Originally from terraform-provider-snowflake/pkg/provider/provider.go
+import (
+	"github.com/chanzuckerberg/go-misc/keypair"
+	"github.com/pkg/errors"
+	"github.com/snowflakedb/gosnowflake"
+)
+
+type SnowflakeConfig struct {
+	Account          string `yaml:"account"`
+	User             string `yaml:"username"`
+	Password         string `yaml:"password"`
+	BrowserAuth      bool   `yaml:"browser_auth"`
+	PrivateKeyPath   string `yaml:"private_key_path"`
+	OauthAccessToken string `yaml:"oauth_access_token"`
+	Region           string `yaml:"region"`
+	Role             string `yaml:"role"`
+}
+
+func ConfigureProvider(s *SnowflakeConfig) (interface{}, error) {
+	dsn, err := DSN(s)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "could not build dsn for snowflake connection")
+	}
+
+	db, err := Open(dsn)
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not open snowflake database.")
+	}
+
+	return db, nil
```

Comment:
```suggestion
	return Open(dsn)
```
might as well just return directly. If you need to wrap, you can wrap inside `Open` func